File: Python Pattern Programs/Numeric Patterns/Pattern 52.py
n = 4  # size
p = 1  # for printing
q = (n * n) + 1

for x in range(n, 0, -1):
    for y in range(1, x + 1):
        print(str(q) + " ", end="")
        q += 1
    q -= 1
    q = q - ((x - 1) * 2)
    print()

File: CSharpPatternPrograms/NumericPatterns/Pattern 52.cs
using System;

class Pattern_146
{
    public static void Main()
     {
       int i;
       int j;
       int k;

       int d1 = 1; // for printing

       int n = 4; //size

       int d2 = (n * n) + 1;
        /* for 2nd part */

        for (i = n;i >= 1;i--)
    {
     
   for (k = 1;k <= i;k++)
     {
      Console.Write(d2+" ");
      d2++;
     }
     d2--; // decrement to get last printed value

     d2 = d2 - ((i - 1) * 2); // used to set next print value

     Console.WriteLine();
    }
          Console.ReadKey(true);
   }
}
